src/Education/SonarRuleIdUriEncoderDecoder.cs

```diff
@@ -36,5 +38,28 @@ public static Uri EncodeToUri(SonarCompositeRuleId compositeRuleId)
 
             return new Uri(link);
         }
+
+        public static bool TryDecodeToCompositeRuleId(Uri uri, out SonarCompositeRuleId compositeRuleId)
```

Comment:
I thought about instead of returning the boolean to just check if the compositeRuleID is null, is there any reason to do it this way? (Just because you suggested it with a return parameter)

Reply:
Personally, I quite like `Try` methods that return `null` if the action can't be performed - it shorter and seems quite logical to me.

However, `bool TryX(input, out output)` is a common pattern in the framework and in C# in general, so some people find it confusing if a method called `TryX` doesn't follow the pattern.